### DISCUSSION TITLE
Security      Insights  Fix inconsistent permissions in Debian package postinst for /etc/grafana

### DISCUSSION THREAD

**Discussion by rda0** [grafana/grafana#37218] [General]
**What would you like to be added**:

I would like permissions to be applied with `chown` in the same fashion (using `find`) as with `chmod` in the [Debian postinst script](https://github.com/grafana/grafana/blob/master/packaging/deb/control/postinst):

```
diff --git a/packaging/deb/control/postinst b/packaging/deb/control/postinst
index 957f8aef3..a1a2a22fc 100755
--- a/packaging/deb/control/postinst
+++ b/packaging/deb/control/postinst
@@ -43,7 +43,7 @@ case "$1" in
   fi
 
 	# configuration files should not be modifiable by grafana user, as this can be a security issue
-	chown -Rh root:$GRAFANA_GROUP /etc/grafana/*
+	find /etc/grafana ! -path /etc/grafana -exec chown -h root:$GRAFANA_GROUP {} ';'
 	chmod 755 /etc/grafana
 	find /etc/grafana -type f -exec chmod 640 {} ';'
 	find /etc/grafana -type d -exec chmod 755 {} ';'

```

**Why is this needed**:

The current `postinst` script applies the file mode `640` to all files (including hidden dotfiles) inside the directory `/etc/grafana`, but it does not change ownership to `root:$GRAFANA_GROUP` on hidden dotfiles.

We use to create a git repositories in configuration directories to track configuration changes and our monitoring system keeps an eye open for any dirty git repositories. If a commit is done as `root`, git creates some files in `/etc/grafana/.git` and after a grafana package update, those files are not accessible anymore to the monitoring user, which is a member of the `$GRAFANA_GROUP` group.

While a workaround for our setup should be trivial and a no problems exist with the default `/etc/grafana` directory structure, I still kindly ask for a solution at the source.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
